system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -190,6 +195,7 @@ this.TopStoriesFeed = class TopStoriesFeed {
     this.spocsPerNewTabs = settings.spocsPerNewTabs;
     this.timeSegments = settings.timeSegments;
     this.domainAffinityParameterSets = settings.domainAffinityParameterSets;
+    this.recsExpireTime = settings.recsExpireTime;
```

Comment:
Hmm, you've mentioned in the comment that

>  that stories don't have an impression older than 90 mins (configurable server-side)

Looks like it doesn't have any server-side configuration involved here, is there anything missing in here or that'd be added in a future patch?

Reply:
Oh, after another look at the code, it does call the `updateSettings` below. Perhaps we should put a guard here in case the server passes down a `unreasonable` value.

Reply:
Done, made sure value can't be lower than default, as that's the main risk i.e. rotating too often.